### PR TITLE
June 2021 grand merge

### PR DESCRIPTION
**Pull request by jarvist**
ToDo:
- [x] Fix Travis :^)
- [x] ImX(freq) #6 
- [x] ReX(freq) #7 
- [x] Multiple phonon modes #8 
- [x] Multiple phonon modes, mobility theories
- [x] Code clean / refactor

### CODE REVIEW

File: src/FeynmanTheory.jl [hunk 1]
```diff
@@ -37,33 +37,58 @@ F(v,w,α)=(3/(4*v))*(v-w)^2-AF(v,w,α)
...
+function feynmanvw(α; v = 0.0, w = 0.0) # v, w defaults
 
-    myf(x) = F(x[1],x[2],α) # Wraps the function so just the two variational params are exposed, so that Optim can call it
-
-    # Now updated to use Optim.jl > 0.15.0 call signature (Julia >0.6 only)
-    res=optimize(OnceDifferentiable(myf, initial; autodiff = :forward), 
-                 lower, upper, initial, Fminbox( BFGS() ) )
-    # specify Optim.jl optimizer. This is doing all the work.
+    # Intial guess for v and w.
+    if v == 0.0 || w == 0.0 # Default values to start with. Generates a random float between 1.0 and 11.0
```
Comment (jarvist) [inline src/FeynmanTheory.jl:50]: I don't think this is great style... I think better would be to keep this function just taking a v and w parameter, then wrap this function with a 'random restarts' or 'checked convergence' function that tries the random points & restarts if convergence is not achieved.

But we can probably sort this out when developing / standardising the multiple phonon branch version!

File: src/OedipusRex.jl [hunk 2]
```diff
@@ -27,3 +26,184 @@ function ReX(v=20,w=20)
     # Oh yikes, and it explodes as u->0 takes it to ->0 in the denominator
 end
 
+"""
+NOTE: This is old code that has been replaced by the full χ function.
```
Comment (jarvist) [inline src/OedipusRex.jl:30]: We should either uncomment or delete this whole function.

**Comment by jarvist** [comment]
^- Travis CI looks quite broken... maybe something needs fixing in the .yml file in the repo root.

**Review by jarvist — COMMENTED** [review]
Starting to look really good. :tada:

File: src/OedipusRex.jl (hunk 2 above)
Comment (Neutrino155) [reply src/OedipusRex.jl:30]: Yeah, I agree. I have just included it for my own benefit for now as I suspect I will actually still use this code for evaluating the chi function at small temperatures as I recall it converging faster than the integrals.

File: src/FeynmanTheory.jl (hunk 1 above)
Comment (Neutrino155) [reply src/FeynmanTheory.jl:50]: That does sound better! I wonder when we constrain v and w in a box, can we also constrain w < v during the optimisation? Would avoid the nasty busy that happens when w >= v!

**Comment by jarvist** [comment]
Ah, Travis is actually working now! (As in, it successfully runs the code, and is failing as one of the tests fails with an exception. The only test in MultipleBranches.)

**Comment by jarvist** [comment]
Ah, 80c2c76 and 99cb9de add many megabytes of PDF plots etc. 
I think these should be removed from this repo (it makes it heavy for anyone cloning to do some development). This will require some git fu, to rewrite history. I think if I rebase this whole branch and drop those commits, it would make most sense. Would you be happy with that @Neutrino155 ?

Are these for your paper? They would fit in that git repo, but also they're likely to be remade several times before submission, so writing a script to generate the data is probably more useful.

**Comment by jarvist** [comment]
You're on fire Brad! 🚀

**Comment by jarvist** [comment]
Great! Everything looks fine. The rest we can 'fix in production' ^_^

**Comment by jarvist** [comment]
@JuliaRegistrator register

**Pull request merged**